internal/utils/poller/poller.go

```diff
@@ -38,14 +37,6 @@ func (p *Poller) Start(ctx context.Context) {
 			// Handle context cancellation.
 			log.Info().Msg("Poller stopped due to context cancellation")
 			return
-		case <-p.quit:
```

Comment:
1. we don't use it method anywhere
2. common pattern is to use ctx.Done() for these things
3. there was small bug: when one finishes working with ticker they should call `.Stop()` (it won't be garbage collected even if there are no more references to it)